eligibility_api/tokens.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+import json
+import logging
+import uuid
+
+from jwcrypto import jwe, jws, jwt, jwk
+from typing import Iterable
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def create_jwk(pem_data):
+    if isinstance(pem_data, str):
+        pem_data = bytes(pem_data, "utf-8")
+
+    return jwk.JWK.from_pem(pem_data)
+
+
+class TokenError(Exception):
+    """Error with API request/response token."""
+
+    pass
+
+
+class RequestToken:
+    """Eligibility Verification API request token."""
+
+    def __init__(
+        self,
+        types: Iterable[str],
+        agency: str,
+        jws_signing_alg: str,
+        client_private_jwk: jwk.JWK,
+        jwe_encryption_alg: str,
+        jwe_cek_enc: str,
+        server_public_jwk: jwk.JWK,
+        sub: str,
+        name: str,
+        issuer: str,
+    ):
```

Comment:
This is where having JWKs actually matters right, inside this class? It seems like the `__init__` could/should use the same raw-data abstraction as `Client`, keep the internal implementation internal to the function.

Reply:
Would simplify the code in `Client` too